pkgs/development/python-modules/schemainspect/default.nix

```diff
@@ -0,0 +1,117 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, fetchpatch
+, pythonOlder
+, sqlalchemy
+, sqlbag
+, setuptools
+, poetry-core
+, pytestCheckHook
+, pytest-xdist
+, pytest-sugar
+, postgresql
+, postgresqlTestHook
+,
+}:
+buildPythonPackage rec {
+  pname = "schemainspect";
+  version = "3.1.1663587362";
+  format = "pyproject";
+  disable = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "djrobstep";
+    repo = pname;
+    # no tags on github, version patch number is unix time.
+    rev = "066262d6fb4668f874925305a0b7dbb3ac866882";
+    hash = "sha256-SYpQQhlvexNc/xEgSIk8L8J+Ta+3OZycGLeZGQ6DWzk=";
+  };
+
+  patches = [
+    # https://github.com/djrobstep/schemainspect/pull/87
+    (fetchpatch
+      {
+        name = "specify_poetry.patch";
+        url = "https://github.com/djrobstep/schemainspect/pull/87.patch";
+        hash = "sha256-/SEmcV9GjjvzfbszeGPkfd2DvYenl7bZyWdC0aI3M4M=";
+      })
+  ];
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+  propagatedBuildInputs = [
+    setuptools # needed for 'pkg_resources'
+    sqlalchemy
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-xdist
+    pytest-sugar
+
+    postgresql
+    postgresqlTestHook
+
+    sqlbag
+  ];
+
+  preCheck = ''
+    export PGUSER="nixbld";
+    export postgresqlEnableTCP=1;
+  '';
+  disabledTests = [
+    # These all fail with "List argument must consist only of tuples or dictionaries":
```

Comment:
From what i gather this is an issue with schemainspect assuming an older version of sqlalchemy? Please note the reason why ignoring these tests is fine, possibly linking to an upstream issue tracking which may indicate when/if this has is fixed.

Reply:
Could we just downgrade the sqlalchemy version for schemainspect and import the patch? Because it doesn't seem likely to me that upstream would upgrade any time soon

Reply:
You could use a `sqlalchemy.overrideAttrs (old: { version = "foobar"; src = old.src.overrideAttrs { rev = foo; hash = bar; }; })` in `migra`, but no in the libs in `python3Packages` because propagated build inputs should be consistent across nixpkgs

Reply:
But if these issues are only in the tests and not in the library itself then it should be good? It requires user testing though

Reply:
Is there a way to use the version number specified in their `pyproject.toml`? (There is none, but that could be changed via a pr)

Reply:
That would be IFD, which is disallowed in nixpkgs. You _could_ add a update script which ensures this is set properly. I think the way forward is to disable the tests and notify upstream, linking to the github issue in nixpkgs